WordPress/Classes/Utility/DisplayableImageHelper.m

```diff
@@ -0,0 +1,184 @@
+#import "DisplayableImageHelper.h"
+
+static const NSInteger FeaturedImageMinimumWidth = 640;
+
+@implementation DisplayableImageHelper
+
+/**
+ Get the url path of the image to display for a post.
+
+ @param dict A dictionary representing a posts attachments from the REST API.
+ @return The url path for the featured image or nil
+ */
++ (NSString *)searchPostAttachmentsForImageToDisplay:(NSDictionary *)attachmentsDict
+{
+    NSArray *attachments = [[attachmentsDict dictionaryForKey:@"attachments"] allValues];
+    if ([attachments count] == 0) {
+        return nil;
+    }
+
+    NSString *imageToDisplay;
+
+    attachments = [self filteredAttachmentsArray:attachments];
+
+    NSDictionary *attachment = [attachments firstObject];
+    NSInteger width = [[attachment numberForKey:@"width"] integerValue];
+    if (width >= FeaturedImageMinimumWidth) {
+        imageToDisplay = [attachment stringForKey:@"URL"];
+    }
+
+    return imageToDisplay;
+}
+
++ (NSArray *)filteredAttachmentsArray:(NSArray *)attachments
+{
+    NSString *key = @"mime_type";
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K BEGINSWITH %@", key, @"image"];
+    attachments = [attachments filteredArrayUsingPredicate:predicate];
+
+    return [attachments sortedArrayUsingComparator:^NSComparisonResult(id obj1, id obj2) {
+        NSDictionary *attachmentA = (NSDictionary *)obj1;
+        NSDictionary *attachmentB = (NSDictionary *)obj2;
+        NSString *key = @"width";
+        NSNumber *widthA = [attachmentA numberForKey:key] ?: @(0);
+        NSNumber *widthB = [attachmentB numberForKey:key] ?: @(0);
+
+        if ([widthA integerValue] < [widthB integerValue]) {
+            return NSOrderedDescending;
+        } else if ([widthA integerValue] > [widthB integerValue]) {
+            return NSOrderedAscending;
+        } else {
+            return NSOrderedSame;
+        }
+    }];
+}
+
+/**
+ Search the passed string for an image that is a good candidate to feature.
+
+ @param content The content string to search.
+ @return The url path for the image or an empty string.
+ */
```

Comment:
Also, consider adding the `@details` tag to explain in more detail what this method is supposed to do (ie: what the outside caller can test against).  I think this method suffers a bit from complexity that may make it not-so-easy to maintain and the documentation should help out a bit on that.